e2e/nomostest/clean.go

```diff
@@ -372,198 +313,207 @@ func resetSystemNamespaces(nt *NT, failOnError FailOnError) {
+	obj.SetName(configuration.Name)
+	return deleteObject(nt, obj)
 }
 
 // removeResourceGroupController deletes namespace `resource-group-system`.
-func removeResourceGroupController(nt *NT, failOnError FailOnError) {
-	_, err := nt.Kubectl("delete", "namespace", "resource-group-system", "--ignore-not-found")
-	if err != nil {
-		if failOnError {
-			nt.T.Fatal("error deleting namespace resource-group-system")
-		} else {
-			nt.T.Log("error deleting namespace resource-group-system")
-		}
-	}
+func removeResourceGroupController(nt *NT) error {
+	obj := &corev1.Namespace{}
+	obj.SetName(configmanagement.RGControllerNamespace)
+	return deleteObject(nt, obj)
 }
 
 func deleteReconcilerManager(nt *NT) error {
 	obj := &appsv1.Deployment{}
 	obj.Namespace = configmanagement.ControllerNamespace
 	obj.Name = reconcilermanager.ManagerName
-	if err := nt.Delete(obj); err != nil {
-		if !apierrors.IsNotFound(err) {
-			return err
-		}
-	}
-	return nil
+	return deleteObject(nt, obj)
 }
 
 func deleteReconcilersBySyncKind(nt *NT, kind string) error {
+	var opts []client.ListOption
+	opts = append(opts, client.InNamespace(configmanagement.ControllerNamespace))
+	opts = append(opts, withLabelListOption(metadata.SyncKindLabel, kind))
 	dList := &appsv1.DeploymentList{}
-	labelSelectorOption := client.MatchingLabelsSelector{
-		Selector: labels.Set{metadata.SyncKindLabel: kind}.AsSelector(),
-	}
-	if err := nt.List(dList, client.InNamespace(configmanagement.ControllerNamespace), labelSelectorOption); err != nil {
+	if err := nt.List(dList, opts...); err != nil {
 		return err
 	}
-	for _, obj := range dList.Items {
-		if err := nt.Delete(&obj); err != nil {
-			if !apierrors.IsNotFound(err) {
-				return err
-			}
+	for _, item := range dList.Items {
+		if err := deleteObject(nt, &item); err != nil {
+			return err
 		}
 	}
 	return nil
 }
 
-func listType(nt *NT, gvk schema.GroupVersionKind) ([]unstructured.Unstructured, error) {
+func listObjectsWithTestLabel(nt *NT, gvk schema.GroupVersionKind) ([]unstructured.Unstructured, error) {
 	list := &unstructured.UnstructuredList{}
 	list.GetObjectKind().SetGroupVersionKind(gvk)
-	var opts []client.ListOption
-	if gvk.Kind != "SyncList" {
-		// For Syncs we can't rely on the TestLabel as this is generated by the
-		// controller. We want all Syncs to be deleted.
-		opts = append(opts, &client.MatchingLabels{TestLabel: TestLabelValue})
-	}
-	err := nt.Client.List(nt.Context, list, opts...)
-	if err != nil && !meta.IsNoMatchError(err) && !apierrors.IsNotFound(err) {
-		// Ignore cases where the type doesn't exist on the cluster. Obviously
-		// there isn't anything to clean in that case.
+	if err := nt.List(list, withLabelListOption(TestLabel, TestLabelValue)); err != nil {
+		if meta.IsNoMatchError(err) {
+			// Resource not registered. So none can exist.
+			return nil, nil
+		}
 		return nil, errors.Wrapf(err, "unable to clean objects of type %v from cluster", gvk)
```

Comment:
Granted this error message was already here, but is this a precise error message? Seems like this error is related to listing rather than cleaning

Reply:
I'll just make it return err.

Reply:
Done.